source/multigrid/mg_transfer_internal.cc

```diff
@@ -98,16 +98,20 @@ namespace internal
       for (unsigned int level = 0; level < n_levels; ++level)
         {
           std::vector<bool> dof_touched(globally_relevant.n_elements(), false);
-          copy_indices[level].clear();
-          copy_indices_level_mine[level].clear();
-          copy_indices_global_mine[level].clear();
 
-          typename dealii::DoFHandler<dim, spacedim>::active_cell_iterator
-            level_cell = mg_dof.begin_active(level);
-          const typename dealii::DoFHandler<dim, spacedim>::active_cell_iterator
-            level_end = mg_dof.end_active(level);
+          // for the most common case where copy_indices are locally owned
+          // both globally and on the level, we want to skip collecting pairs
+          // and later sorting them. instead, we insert these indices into a
+          // plain vector
+          std::vector<types::global_dof_index> unrolled_copy_indices;
 
-          for (; level_cell != level_end; ++level_cell)
+          copy_indices_level_mine[level]  = {};
```

Comment:
I find `.clear();` much more often than `= {}`. Is there any reason for choosing this syntax?

Reply:
`std::vector::clear()` does not release previously allocated memory, so I could in principle imagine a case where we would hold more data from a previous round. This variable can be passed on from a member variable e.g. in `MGTransferPrebuilt`. However, that is not always the case, e.g. not for `MGTransferMatrixFree` which builds this from temporary data, so let us stick with the more common `clear()` and remove this micro optimization.

Reply:
I expected `= {}` to not release memory either. At least in https://godbolt.org/z/qOH8-v, I see that the capacity is the same after the copy.

Reply:
Alright, thanks for confirming. One more reason to stick to the old code.